packages/web3-core-method/src/index.js

```diff
@@ -653,7 +653,15 @@ Method.prototype.buildCall = function () {
                 if (!err && method.isRevertReasonString(result)){
                     reasonData = result.substring(10);
                 } else if (err && err.data){
-                    reasonData = (err.data.data || err.data).substring(10);
+                    // workaround embedded error details got from some providers like MetaMask
+                    if (typeof err.data === 'object') {
+                        // Ganache has no `originalError` sub-object unlike others
+                        var originalError = err.data.originalError ?? err.data;
```

Comment:
yes, you are right, the revert data is either in `err.data` (original code until yesterday and your PR), `err.data.data` or `err.data.originalError.data`, see [my comment](https://github.com/web3/web3.js/issues/4454#issuecomment-1485953455).
That's why I divided the code in two parts so that L660:
`reasonData = <originalError = err.data | err.data.originalError >.data.substring(10)`
Is it ok @Muhammad-Altabba  ?